src/com/palmergames/bukkit/towny/object/economy/Account.java

```diff
@@ -0,0 +1,320 @@
+		observers.add(GLOBAL_OBSERVER);
+	}
+	
+	// Template methods
+	protected abstract boolean addMoney(double amount);
+	protected abstract boolean subtractMoney(double amount);
+
+	/**
+	 * Attempts to add money to the account, 
+	 * and notifies account observers of any changes.
+	 * 
+	 * @param amount The amount to add.
+	 * @param reason The reason for adding.
+	 * @return boolean indicating success.
+	 * @throws EconomyException On an economy error.
+	 */
+	public boolean deposit(double amount, String reason) throws EconomyException {
+		if (TownySettings.getBoolean(ConfigNodes.ECO_CLOSED_ECONOMY_ENABLED)) {
+			return payFromServer(amount, reason);
+		}
+		if (addMoney(amount)) {
+			notifyObserversDeposit(this, amount, reason);
+			return true;
+		}
+		
+		return false;
+	}
+
+	/**
+	 * Attempts to withdraw money from the account, 
+	 * and notifies account observers of any changes.
+	 *
+	 * @param amount The amount to withdraw.
+	 * @param reason The reason for subtracting.
+	 * @return boolean indicating success.
+	 * @throws EconomyException On an economy error.
+	 */
+	public boolean withdraw(double amount, String reason) throws EconomyException {
+		if (TownySettings.getBoolean(ConfigNodes.ECO_CLOSED_ECONOMY_ENABLED)) {
+			return payToServer(amount, reason);
+		}
+		if (subtractMoney(amount)) {
+			notifyObserversWithdraw(this, amount, reason);
+			return true;
+		}
+		
+		return false;
+	}
+
+	/**
+	 * Pays another account the specified funds.
+	 *
+	 * @param amount The amount to pay.
+	 * @param collector The account to pay.
+	 * @param reason The reason for the pay. 
+	 * @return boolean indicating success.
+	 * @throws EconomyException On an economy error.
+	 */
+	public boolean payTo(double amount, EconomyHandler collector, String reason) throws EconomyException {
+		return payTo(amount, collector.getAccount(), reason);
+	}
+	
+	protected boolean payToServer(double amount, String reason) throws EconomyException {
+		// Take money out.
+		withdraw(amount, reason);
+		
+		// Put it back into the server.
+		return TownyEconomyHandler.addToServer(amount, getBukkitWorld());
+	}
+	
+	protected boolean payFromServer(double amount, String reason) throws EconomyException {
+		// Put money in.
+		deposit(amount, reason);
+		
+		// Remove it from the server economy.
+		return TownyEconomyHandler.subtractFromServer(amount, getBukkitWorld());
+	}
+
+	/**
+	 * Pays another account the specified funds.
+	 *
+	 * @param amount The amount to pay.
+	 * @param collector The account to pay.
+	 * @param reason The reason for the pay.
+	 * @return boolean indicating success.
+	 * @throws EconomyException On an economy error.
+	 */
+	public boolean payTo(double amount, Account collector, String reason) throws EconomyException {
+		
+		if (amount > getHoldingBalance()) {
+			throw new EconomyException("Not enough money");
+		}
+		
+		if (!withdraw(amount, reason)) {
+			return false;
+		}
+
+		return collector.deposit(amount, reason);
```

Comment:
Unless there is further logic planned to be added in that if-statement, can probably be simplified to `return withdraw(...) && collector.deposit(...):`